tests/haskell_module/single/test.bzl

```diff
@@ -6,24 +6,21 @@ def _default_info_test_impl(ctx):
     target_under_test = analysistest.target_under_test(env)
 
     default_info = target_under_test[DefaultInfo]
+    actual_files = [x.basename for x in default_info.files.to_list()]
 
-    obj_files = [
-        f
-        for f in default_info.files.to_list()
-        if paths.split_extension(f.path)[1] == ".o"
-    ]
-    asserts.true(env, len(obj_files) == 1, "Expected exactly one object file")
-    [obj_file] = obj_files
-    asserts.equals(env, "Single.o", obj_file.basename)
+    # can't use is_profiling_enabled(haskell_context(ctx)) here, because we
+    # don't have access to the haskell toolchain
+    with_profiling = ctx.var["COMPILATION_MODE"] == "dbg"
 
-    hi_files = [
-        f
-        for f in default_info.files.to_list()
-        if paths.split_extension(f.path)[1] == ".hi"
-    ]
-    asserts.true(env, len(hi_files) == 1, "Expected exactly one interface file")
-    [hi_file] = hi_files
-    asserts.equals(env, "Single.hi", hi_file.basename)
+    if with_profiling:
+        expected_files = ["Single.p_o", "Single.p_hi"]
+    else:
+        expected_files = ["Single.o", "Single.hi"]
+
+    for f in expected_files:
+        asserts.true(env, f in actual_files, "Expected file %s not produced" % f)
+    for f in actual_files:
+        asserts.true(env, f in expected_files, "Found unexpected file %s" % f)
```

Comment:
Perhaps this could be simplified using [`asserts.set_equals`](https://github.com/bazelbuild/bazel-skylib/blob/fd75066f159234265efb8f838b056be5a2e00a59/lib/unittest.bzl#L502)?

Reply:
It could, but I thought the more precise error message this gives (exactly which files are / are not present) would be more useful.

I suppose I could rephrase the error message to "Exepected files: blah. Got files: blah instead"

I'm happy to change it if you think that's better.

Reply:
> It could, but I thought the more precise error message this gives (exactly which files are / are not present) would be more useful.

That is useful, indeed. [Looking at the code](https://github.com/bazelbuild/bazel-skylib/blob/fd75066f159234265efb8f838b056be5a2e00a59/lib/unittest.bzl#L423-L426), though, `asserts.set_equals` should produce that kind of message.

Reply:
Good catch. switched it to using `asserts.set_equals`

Reply:
Thank you! In that case the sets need to be first constructed with [`sets.make`](https://github.com/bazelbuild/bazel-skylib/blob/fd75066f159234265efb8f838b056be5a2e00a59/lib/sets.bzl)

Reply:
Ah, thank you. I saw that error on CI. For some reason it's not happening when I run the test locally :eyes: